mongo/errors.go

```diff
@@ -112,6 +113,23 @@ func (e MongocryptdError) Unwrap() error {
 	return e.Wrapped
 }
 
+// ServerError is the interface implemented by errors returned from the server. Custom implementations of this
+// interface should not be used in production.
```

Comment:
IIUC `serverError` enforces that a user of the mongo package can't implement the ServerError interface. I was thinking it could say something like: "ServerError cannot be implemented" if that seems like a reasonable thing to note.

Reply:
I'm not sure how necessary that is, because a user will figure it out pretty fast if they try. This disclaimer is taken from the Session interface, which is also unimplementable.